src/main/java/com/redhat/darcy/salesforce/DateInputField.java

```diff
@@ -0,0 +1,152 @@
+
+import com.redhat.darcy.ui.AbstractViewElement;
+import com.redhat.darcy.ui.annotations.Require;
+import com.redhat.darcy.ui.api.Locator;
+import com.redhat.darcy.ui.api.elements.DateInput;
+import com.redhat.darcy.ui.api.elements.Element;
+import com.redhat.darcy.ui.api.elements.Link;
+import com.redhat.darcy.ui.api.elements.Requireable;
+import com.redhat.darcy.ui.api.elements.TextInput;
+
+import static com.redhat.darcy.salesforce.RequiredInput.requiredInput;
+import static com.redhat.darcy.ui.By.xpath;
+import static com.redhat.darcy.ui.Elements.link;
+import static com.redhat.darcy.ui.Elements.textInput;
+import static com.redhat.darcy.web.By.htmlTag;
+
+import java.time.LocalDate;
+
+/**
+ * An ViewElement for a value that corresponds to a Date field on 
+ * a Salesforce object.
+ */
+public class DateInputField extends AbstractViewElement implements TextInput, 
+    DateInput, Link, Requireable {
+
+    @Require
+    private Element parent = super.parent;
+    
+    @Require
+    private TextInput nestedTextInput = textInput(byInner( 
+            xpath("//span[contains(@class,'dateInput')]"),
+            htmlTag("input")));
+    
+    @Require
+    private Link nestedLink = link(byInner(
+            xpath("//span[contains(@class,'dateInput')]"),
+            htmlTag("span"), htmlTag("a")));
+    
+    
+    private RequiredInput requiredInput;
+    
+    /**
+     * A ViewElement that corresponds to a Date field on
+     * a Salesforce object.  Takes the locator returned from BySalesforce and 
+     * finds the elements for the date text input and the [today] link.
+     * 
+     * @param locator  Locator returned from BySalesforce
+     * @return DateInputField
+     */
+    public static DateInputField dateInputField(Locator locator) {
+        return new DateInputField(locator);
+    }
+
+    public DateInputField(Locator parent) {
+        super(parent);
+        requiredInput = requiredInput(parent);
+
+    }
+
+    @Override
+    public void clearAndType(CharSequence... keysToType) {
+        nestedTextInput.clearAndType(keysToType);
+    }
+
+    @Override
+    public void type(CharSequence... keysToType) {
+        nestedTextInput.type(keysToType);
+    }
+
+    @Override
+    public void clear() {
+        nestedTextInput.clear();
+    }
+    
+    /** Click the date input field.  @see today() for the [today] link. */
+    @Override
+    public void click() {
+        nestedTextInput.click();
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return nestedTextInput.isEnabled();
+    }
+
+    @Override
+    public boolean isDisplayed() {
+        return nestedTextInput.isDisplayed();
+    }
+
+    @Override
+    public boolean isPresent() {
+        return nestedTextInput.isPresent();
+    }
+
+    @Override
+    public String getValue() {
+        return nestedTextInput.getValue();
+    }
+
+    @Override
+    public void setDate(LocalDate date) {
+        nestedTextInput.clearAndType(date.toString());
```

Comment:
No, I was delegating that to Salesforce.  It will give a validation error if you use an invalid format. What constitutes a valid format also depends on the context of the user (Country specific date formats) who is entering the data.